File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/houses/HouseSellRequestMessage.scala





















// Generated on 05/03/2013 10:34:44
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait HouseSellRequestMessage extends MessageDeserializer {
  val messageId = 5697

  protected trait Message extends BaseMessage with Serializable {
    val messageId = HouseSellRequestMessage.messageId

    val amount: Int

    def serialize(buf: Buffer) {
      buf.writeInt(amount)
    }

  }

  protected def create(amount: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val amount = buf.readInt()

    create(amount)
  }

}

object HouseSellRequestMessage extends HouseSellRequestMessage {
  case class MessageImpl(amount: Int) extends Message
  type Target = MessageImpl

  override def create(amount: Int) = MessageImpl(amount)

  def apply(amount: Int) = create(amount)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/HouseSellRequestMessage.java


// Generated on 05/08/2013 19:37:46
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class HouseSellRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5697;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int amount;
    
    public HouseSellRequestMessage() { }
    
    public HouseSellRequestMessage(int amount) {
        this.amount = amount;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(amount);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        amount = buf.readInt();
        if (amount < 0)
            throw new RuntimeException("Forbidden value on amount = " + amount + ", it doesn't respect the following condition : amount < 0");
    }
    
}
